libdocker/kube_docker_client.go

```diff
@@ -362,22 +359,20 @@ func (p *progressReporter) start() {
 	go func() {
 		ticker := time.NewTicker(defaultImagePullingProgressReportInterval)
 		defer ticker.Stop()
+		var lastTimeStamp time.Time
 		for {
 			select {
 			case <-ticker.C:
 				progress, timestamp := p.progress.get()
-				// If there is no progress for p.imagePullProgressDeadline, cancel the operation.
 				if time.Since(timestamp) > p.imagePullProgressDeadline {
```

Comment:
```suggestion
				if progress.Message.Status != "Extracting" && time.Since(timestamp) > p.imagePullProgressDeadline {
```
You don't need any other changes.

Reply:
Thanks for the advice. `progress` here is a string, shall we change it to `strings.Contains(progress, "Extracting")` ?

Reply:
My mistake, I didn't notice that `p.progress.get()` retuned a string formatted from fields of the JSONMessage. When the image is being extracted, the `(dockermessage. JSONMessage).Status == "Extracting"`, so it would be silly to test for that case by checking whether "Extracting" is contained within a string that cri-dockerd itself formatted.